packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx

```diff
@@ -385,14 +411,18 @@ export class Dropdown extends BaseComponent<IDropdownInternalProps, IDropdownSta
         break;
 
       case KeyCodes.up:
-        this.setSelectedIndex(this.state.selectedIndex - 1);
+        newIndex = this._moveIndex(this.state.selectedIndex - 1);
```

Comment:
To simplify this logic, why not just do this:

```tsx
this._moveIndex(ev, -1);
return;
```
Then make move index do the work of moving in that direction;

```tsx
private _moveIndex(directionStep, ev) {
  let currentIndex = this.state.selectedIndex + directionStep;

  // Iterate to the next selectable thing.
  while (currentIndex >= 0 && currentIndex <= options.length && options[currentIndex].itemType !== Normal) {
    currentIndex += directionStep;
  }

  // If thing is different than state and selectable...
  if (currentIndex !== this.state.selectedIndex && currentIndex >= 0 && currentIndex < options.length && options[currentIndex].itemType === Normal) {

    this.setSelectedIndex(currentIndex);
    ev.stopPropagation();
    ev.preventDefault();
  }
}